cinema/views.py

```diff
@@ -43,16 +70,82 @@ def get_serializer_class(self):
 
         return MovieSerializer
 
+    def get_queryset(self):
+        queryset = super().get_queryset()
+
+        title = self.request.query_params.get("title", None)
+        actors = self.request.query_params.get("actors", None)
+        genres = self.request.query_params.get("genres", None)
+
+        if title:
+            queryset = queryset.filter(title__icontains=title)
+
+        if genres:
+            genres_ids = [int(str_id) for str_id in genres.split(",")]
+            queryset = queryset.filter(genres__in=genres_ids)
+
+        if actors:
+            actors_ids = [int(str_id) for str_id in actors.split(",")]
+            queryset = queryset.filter(actors__in=actors_ids)
+
+        return queryset.distinct()
+
+    def paginate_queryset(self, queryset):
+        return None
```

Comment:
The `paginate_queryset` method in `MovieViewSet` returns `None`, which disables pagination. If pagination is required, consider using the default pagination or a custom pagination class instead of returning `None`.